composer/helpers/v2/src/UpdateChecker.php

```diff
@@ -48,10 +50,22 @@ public static function getLatestResolvableVersion(array $args): ?string
             $io->loadConfiguration($config);
         }
 
+        $package = $composer->getPackage();
+
+        $versionParser = new VersionParser();
+
+        $constraint = $versionParser->parseConstraints($latestAllowableVersion); // your version constraint
+        $packageLink = new Link($package->getName(), $dependencyName, $constraint);
+
+        $requires = $package->getRequires();
+        $requires[$dependencyName] = $packageLink;
+
+        $package->setRequires($requires);
```

Comment:
I don't fully understand what is going on here, but it looks like you are **overriding** the constraint from the `composer.json` file with `$latestAllowableVersion`. I don't know what `$latestAllowableVersion` contains, but at a guess it looks like the latest version obtained from the registry? Should it not combine that with the existing constraint instead of overriding it? Does this work properly with dev constraints or is this going to introduce a non-dev constraint when updating dev requirements?

I suspect this is probably the cause of #10033